packages/flutter/lib/src/painting/borders.dart

```diff
@@ -30,16 +30,91 @@ enum BorderStyle {
 /// - [inside] provides padding with full [BorderSide.width].
 /// - [center] provides padding with half [BorderSide.width].
 /// - [outside] provides zero padding, as stroke is drawn entirely outside.
-enum StrokeAlign {
+@immutable
+class StrokeAlign {
+  /// This constructor allows any value between 0.0 and 1.0.
+  const StrokeAlign(this.value)
+      : assert(value != null),
+        assert(value >= 0.0, 'StrokeAlign must be greater than or equal to 0.0.'),
+        assert(value <= 1.0, 'StrokeAlign must be less than or equal to 1.0.');
+
   /// The border is drawn on the inside of the border path.
-  inside,
+  static const StrokeAlign inside = StrokeAlign(0.0);
 
   /// The border is drawn on the center of the border path, with half of the
   /// [BorderSide.width] on the inside, and the other half on the outside of the path.
-  center,
+  static const StrokeAlign center = StrokeAlign(0.5);
 
   /// The border is drawn on the outside of the border path.
-  outside,
+  static const StrokeAlign outside = StrokeAlign(1.0);
+
+  /// Mimicks enum.values.
+  static const List<StrokeAlign> values = <StrokeAlign>[inside, center, outside];
+
+  /// Mimicks enum.name.
+  /// Returns the current name for StrokeAlign, if it has one.
+  String get name {
+    if (this == inside) {
+      return 'inside';
+    } else if (this == center) {
+      return 'center';
+    } else if (this == outside) {
+      return 'outside';
+    } else {
+      return '$value';
+    }
+  }
+
+  /// A value representing StrokeAlign between 0.0 and 1.0.
+  final double value;
+
+  @override
+  String toString() => 'StrokeAlign($value)';
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is StrokeAlign
+        && other.value == value;
+  }
+
+  @override
+  int get hashCode => value.hashCode;
+
+  /// Lerp a side contained in a circle.
+  static double radiusLerp(Rect rect, BorderSide side) {
+    if (side.strokeAlign.value < 0.5) { // inside
+      return ui.lerpDouble((rect.shortestSide - side.width) / 2.0, rect.shortestSide / 2.0, 2 * side.strokeAlign.value)!;
+    } else if (side.strokeAlign.value == 0.5) {
+      return rect.shortestSide / 2.0; // center
+    } else { // outside
+      return ui.lerpDouble(rect.shortestSide / 2.0, (rect.shortestSide + side.width) / 2.0, 2 * side.strokeAlign.value - 0.5)!;
+    }
+  }
+
+  /// Lerp a side contained in a [RRect].
+  static RRect rrectLerp(RRect rrect, BorderSide side) {
+    if (side.strokeAlign.value < 0.5) { // inside
+      return rrect.deflate(ui.lerpDouble(side.width / 2, 0, 2 * side.strokeAlign.value)!);
+    } else if (side.strokeAlign.value == 0.5) { // center
+      return rrect;
+    } else { // outside
+      return rrect.inflate(ui.lerpDouble(0, side.width / 2, 2 * side.strokeAlign.value - 0.5)!);
+    }
+  }
```

Comment:
Similarly, this can be:

```suggestion
  static RRect rrectLerp(RRect rrect, BorderSide side) {
    return rrect.inflate(side.width * side.strokeAlign.value);
  }
```

(the inflate arg can be negative).

Reply:
wowwww I never thought about this 😳